analyzer/src/main/kotlin/managers/Gradle.kt

```diff
@@ -154,28 +157,50 @@ class Gradle : PackageManager() {
 
     private fun parseDependency(dependency: Dependency, packages: MutableMap<String, Package>,
                                 repositories: List<RemoteRepository>): PackageReference {
+        val errors = dependency.error?.let { mutableListOf(it) } ?: mutableListOf()
```

Comment:
How about combining this with the `if` below where you also check `dependency.error` for `null`, and let the `if` return either `mutableListOf()` or else `mutableListOf(dependency.error)`.